app/uk/gov/hmrc/fileupload/controllers/envelope.scala

```diff
@@ -47,21 +49,51 @@ object EnvelopeReport {
       expiryDate = envelope.expiryDate,
       status = Some(envelope.status.name),
       metadata = envelope.metadata,
+      constraints = envelope.constraints,
       destination = envelope.destination,
       application = envelope.application,
       files = fileReports
     )
   }
-
 }
 
 case class CreateEnvelopeRequest(callbackUrl: Option[String] = None,
                                  expiryDate: Option[DateTime] = None,
-                                 metadata: Option[JsObject] = None)
+                                 metadata: Option[JsObject] = None,
+                                 constraints: Option[EnvelopeConstraintsO] = None)
+
+case class EnvelopeConstraintsO(maxNumFiles: Option[Int] = None,
```

Comment:
That's not how we deal with user configuration parameters that also have default values, we don't create a duplication of the case classes for that @kastoestoramadus . Please see other examples in our codebase and apply a similar approach.